source/common/http/http2/codec_impl.cc

```diff
@@ -1841,8 +1841,13 @@ ConnectionImpl::Http2Callbacks::Http2Callbacks(bool use_new_codec_wrapper) {
 ConnectionImpl::Http2Callbacks::~Http2Callbacks() { nghttp2_session_callbacks_del(callbacks_); }
 
 ConnectionImpl::Http2Options::Http2Options(
-    const envoy::config::core::v3::Http2ProtocolOptions& http2_options) {
+    const envoy::config::core::v3::Http2ProtocolOptions& http2_options,
+    const bool disable_header_validation) {
```

Comment:
@yanavlasov I wasn't sure how to export this setting; I can remove this commit and only keep the nghttp2 until we have a `enable_unified_header_validation` setting in the public interface

Reply:
This looks very good. For this PR, let's plumb the `disable_header_validation` parameter up to the `ServerConnectionImpl` and `ClientConnectionImpl` constructors. Then it would be possible to add tests to the http2/codec_impl_test.cc that captures the original and new behavior based on this flag.